ext/pdo_mysql/tests/gh15093.phpt

```diff
@@ -0,0 +1,170 @@
+--TEST--
+GH-15093: Add support for all flags from mysql in the PDO MySql driver in the getColumnMeta function.
+--EXTENSIONS--
+pdo_mysql
+--SKIPIF--
+<?php
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+MySQLPDOTest::skip();
+?>
+--FILE--
+<?php
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+$db = MySQLPDOTest::factory();
+
+$db->exec("
+    CREATE TABLE `gh_15093` (
+        `id` INT NOT NULL AUTO_INCREMENT,
+        `uuid` BINARY(16) DEFAULT (uuid_to_bin(uuid())),
+        `blob` BLOB,
+        `ts` TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+        `set` SET('one', 'two'),
+        `enum` ENUM('a', 'b', 'c'),
+        `num` INT(11) UNSIGNED ZEROFILL DEFAULT 0,
+        PRIMARY KEY(`id`),
+        UNIQUE KEY `UUID` (`uuid`)
+    )
+");
+
+$stmt = $db->prepare('SELECT `id`, `uuid`, `blob`, `ts`, `set`, `enum`, `num` FROM gh_15093');
+$stmt->execute();
+
+$n = $stmt->columnCount();
+$meta = [];
+
+for ($i = 0; $i < $n; ++$i) {
+    $m = $stmt->getColumnMeta($i);
+
+    // libmysql and mysqlnd will show the pdo_type entry at a different position in the hash
+    // and will report a different type, as mysqlnd returns native types.
+    unset($m['pdo_type']);
```

Comment:
note: that or that might be ways around it. e.g. sorting.

Reply:
This section was pulled from another test. As stated in the comment, the value of the pdo_type may differ based on which driver is used. I only modify the flags and added a native_flags attribute, which is really what the test is meant to cover.

Reply:
fair enough, anyhow the maintainer will review all properly soon :)

Reply:
Is it possible to split the tests per driver and check the value of `pdo_type` in the test to ensure I am not breaking existing functionality?
These behaviors don't change from driver to driver, just the values ​​change, so I think it's okay to just check it with mysqlnd testing.